### DISCUSSION TITLE
Template rendering is too slow

### DISCUSSION THREAD

**Discussion by Nimaj1994** [ckan/ckan#7847] [General]
## CKAN version
2.10
## Describe the bug
when ever I open dataset page it takes more than 20 seconds but in that time I can get more than 250 packages details by api.

### Steps to reproduce
just click on dataset page

### Expected behavior
take less than 2 seconds to response

### Additional details
If possible, please provide the full stack trace of the error raised,  or add screenshots to help explain your problem.

**Comment by f-sardivand** [comment]
@Nimaj1994 yes I have this problem, if it's possible give me a solution

**Comment by pdelboca** [comment]
Hi @Nimaj1994 ad @f-sardivand !

Can you please provide more context on your issue? I currently have a local instance for development with 800+ dataset and I'm not experience this issue.

Could you provide:
 - A screenshot of the network calls.
 - Some screenshots of the profiling provided by the FlaskDebugToolbar? Sorted by PerCall(ms) and CumulativeCall(ms)
 - More detail on the Hardware you are running
 - More detail on the stack of plugins you have in your implementations.
 
 Thanks!

**Comment by wardi** [comment]
Also
- how many resources do you have in this dataset?

**Comment by Nimaj1994** [comment]
This issue was due to a change in datasets list template by myself which made internal loop that caused the low performance

**Comment by octavportar** [comment]
> Also
> 
> * how many resources do you have in this dataset?
Hi! I have the same problem on CKAN 
![image](https://github.com/ckan/ckan/assets/79480476/b300f4ab-6e91-46c4-92a3-6290ac5a28f6)
There are 215 resources in this dataset.

**Reply by wardi** [reply]
@PortarescuOctavian I believe your case is caused by #6146 if we can fix our snippet performance your dataset pages (even ones with hundreds of resources) will render much more quickly.
